### PR TITLE
Gpu usage added

### PR DESCRIPTION
**Pull request by georghildebrand**

### CODE REVIEW

**Comment by georghildebrand** [comment]
@elezar , please have a look. i think there is not so much more information needed.

File: components/taupage.rst [hunk 1]
```diff
@@ -688,3 +688,25 @@ You can build your own Taupage AMI using the code from the repository on GitHub
 In the repository you will find a configuration (config-stups-example.sh) file which you'll have to adjust to your needs.
 
 See :ref:`taupage-ami-creation` for details.
+
+
+Using Nvidia gpus with nvidia-docker
++++++++++++++++++++++
```
Comment (hjacobs) [inline components/taupage.rst]: please extend the underline

File: components/taupage.rst [hunk 2]
```diff
@@ -688,3 +688,25 @@ You can build your own Taupage AMI using the code from the repository on GitHub
...
+If gpu device files are found on the host the usage of nvidia-docker is triggered automatically. This has the benefit that drivers are automatically mounted into the docker container (there is no need to install identical drivers in the container).
+
+Some requirements:
+
+  * gpu instances must be available in the zone of your stups team account (eg: eu-west-1)
```
Comment (hjacobs) [inline components/taupage.rst]: * gpu -> GPU
* stups -> STUPS

the list indentation looks wrong, did you test with a local HTML build?

**Comment by hjacobs** [comment]
@georghildebrand please check the HTML rendering locally by running `make html` --- I suspect some `.rst` formatting to be wrong.

File: components/taupage.rst [hunk 3]
```diff
@@ -688,3 +688,25 @@ You can build your own Taupage AMI using the code from the repository on GitHub
 In the repository you will find a configuration (config-stups-example.sh) file which you'll have to adjust to your needs.
 
 See :ref:`taupage-ami-creation` for details.
+
+
+Using Nvidia gpus with nvidia-docker
++++++++++++++++++++++
+If gpu device files are found on the host the usage of nvidia-docker is triggered automatically. This has the benefit that drivers are automatically mounted into the docker container (there is no need to install identical drivers in the container).
```
Comment (hjacobs) [inline components/taupage.rst]: nit: gpu -> GPU

Comment (hjacobs) [inline components/taupage.rst]: nit: docker -> Docker

Comment (mikkeloscar) [reply components/taupage.rst]: -> Moby ^^

Comment (hjacobs) [reply components/taupage.rst]: Oh no, I just started reading the PR and Hacker News thread... Docker (the company) is unbelievable... (not in a positive sense)

**Comment by elezar** [comment]
@hjacobs @mikkeloscar I have created georghildebrand/stups-documentation#1 (as I didn't want to hijack this PR) to address your comments and add some edits. One question that I did have is where to add an example. It seems to me as if the Taupage documentation is not necessarily the correct place for this. What are your thoughts on this?

(I could also create a PR to replace this one)

**Comment by mikkeloscar** [comment]
@elezar putting an example under the user-guide section makes sense to me.

**Comment by georghildebrand** [comment]
@elezar, thanks for you PR and the corrections. @hjacobs , i merged evans suggestions.

**Comment by georghildebrand** [comment]
@elezar, do you want to take care of creating an example for the user guide? We should create an issue for that...

**Comment by hjacobs** [comment]
:+1:

File: components/taupage.rst [hunk 4]
```diff
@@ -688,3 +688,18 @@ You can build your own Taupage AMI using the code from the repository on GitHub
 In the repository you will find a configuration (config-stups-example.sh) file which you'll have to adjust to your needs.
 
 See :ref:`taupage-ami-creation` for details.
+
+Support for NVIDIA GPUs
++++++++++++++++++++++++
+
+Taupage supports the use of NVIDIA CUDA-enabled GPUs if these are available on the EC2 host (e.g. G2 or P2 instances). In this case, `nvidia-docker` (https://github.com/NVIDIA/nvidia-docker) is used as a drop-in replacement for Docker. This creates a Docker volume which contains the CUDA driver files installed on the host. This volume, as well as the required NVIDIA device nodes, are mounted into the running container allowing GPU-enabled applications to be run.
```
Comment (hjacobs) [inline components/taupage.rst]: hyperlink the text instead of having the URL in the text: http://www.sphinx-doc.org/en/stable/rest.html#hyperlinks

Comment (georghildebrand) [reply components/taupage.rst]: done

**Comment by hjacobs** [comment]
:+1:

**Comment by elezar** [comment]
@georghildebrand I will setup the example and create another PR to this branch in your repo.

**Comment by georghildebrand** [comment]
@elezar, thank your for the hello world example. @mikkeloscar, do you have rights to move evans example git: https://github.com/elezar/gpu-hello-world to our stups projects?

**Comment by mikkeloscar** [comment]
@georghildebrand Yes I can move it.

**Comment by mikkeloscar** [comment]
I have moved it now: https://github.com/zalando-stups/gpu-hello-world

**Comment by hjacobs** [comment]
:+1:

**Comment by georghildebrand** [comment]
:+1:

**Pull request merged**